libraries/AP_GyroFFT/AP_GyroFFT.h

```diff
@@ -185,25 +267,35 @@ class AP_GyroFFT
     uint8_t _current_sample_mode : 3;
     // harmonic multiplier for two highest peaks
     float _harmonic_multiplier;
-    // searched harmonics - inferred from harmonic notch harmoncis
+    // searched harmonics - inferred from harmonic notch harmonics
     uint8_t _harmonics;
+    // engine health
+    uint8_t _health;
 
     // smoothing filter on the output
-    LowPassFilterFloat _center_freq_filter[XYZ_AXIS_COUNT];
+    MedianLowPassFilter3dFloat _center_freq_filter[FrequencyPeak::MAX_TRACKED_PEAKS];
+    // smoothing filter on the energy
+    MedianLowPassFilter3dFloat _center_freq_energy_filter[FrequencyPeak::MAX_TRACKED_PEAKS];
     // smoothing filter on the bandwidth
-    LowPassFilterFloat _center_bandwidth_filter[XYZ_AXIS_COUNT];
+    MedianLowPassFilter3dFloat _center_bandwidth_filter[FrequencyPeak::MAX_TRACKED_PEAKS];
+    // smoothing filter on the frequency fit
+    LowPassFilterFloat _harmonic_fit_filter[XYZ_AXIS_COUNT];
 
     // configured sampling rate
     uint16_t _fft_sampling_rate_hz;
     // number of cycles without a detected signal
-    uint8_t _missed_cycles;
+    uint8_t _missed_cycles[XYZ_AXIS_COUNT][FrequencyPeak::MAX_TRACKED_PEAKS];
+    // number of cycles without a detected signal
+    uint8_t _distorted_cycles[XYZ_AXIS_COUNT];
     // whether the analyzer initialized correctly
     bool _initialized;
 
     // whether the analyzer should be run
     bool _analysis_enabled = true;
     // whether the update thread was created
     bool _thread_created = false;
```

Comment:
init of vars to false (zero) just uses flash space for no reason, all class vars start as zero in ArduPilot

Reply:
ok

Reply:
Fixed